examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -440,16 +485,59 @@ def _cumulated_claims(y_true, y_pred, exposure):
 ax.legend(loc="upper left")
 
 ##############################################################################
-# This plot reveals that the random forest model is slightly better at ranking
-# policyholders by risk profiles even if the absolute value of the predicted
-# expected frequencies are less well calibrated than for the linear Poisson
-# model.
+# As expected, the dummy regressor is unable to discriminate and therefore
+# performs the worst on this plot.
+#
+# The tree-based model is significantly better at ranking policyholders by risk
+# while the two linear models perform similarly.
 #
 # All three models are significantly better than chance but also very far from
 # making perfect predictions.
 #
 # This last point is expected due to the nature of the problem: the occurrence
 # of accidents is mostly dominated by circumstantial causes that are not
 # captured in the columns of the dataset or that are indeed random.
+#
+# The linear models assume no interactions between the input variables which
+# likely causes under-fitting. Inserting a polynomial feature extractor
+# (func:`sklearn.preprocessing.PolynomialFeatures`) indeed increases their
+# discrimative power by 2 points of Gini index. In particular it improves the
+# ability of the models to identify the top 5% riskiest profiles.
+#
+# Main takeaways
+# --------------
+#
+# - A ideal model is both well-calibrated and discriminative.
+#
+# - The Gini index reflects the ability of a model to rank predictions
+#   irrespective of their absolute values, and therefore only assess their
+#   discriminative power.
+#
+# - The calibration of the model can be assessed by plotting the mean observed
+#   value vs the mean predicted value on groups of test samples binned by
+#   predicted risk.
+#
+# - The least squares loss of the Ridge regression model seem to cause this
+#   model to be badly calibrated. In particular it tends to under estimate the
+#   risk and can even predict invalid negative frequencies...
```

Comment:
`...`?

Reply:
Also, it's not the squared error, but rather the linearity assumptions that cause negative predictions.

Reply:
The tree-based model does not make the linear assumption between `E[y]` and `X` and can still predict negative values. But I will mention the implicit identity link function of `Ridge`.